### ISSUE TITLE
Validation loss increases while validation accuracy is still improving

### ISSUE THREAD

**Issue by fish128** [fchollet/keras#3755]
I am training a deep CNN (4 layers) on my data. I used "categorical_crossentropy" as the loss function.
During training, the training loss keeps decreasing and training accuracy keeps increasing until convergence.  But the validation loss started increasing while the validation accuracy is still improving. 
The curves of loss and accuracy are shown in the following figures:
![acc_cnn32p-32p-32p-32p_adam1e-4_5000](https://cloud.githubusercontent.com/assets/8138843/18455543/3b15fc32-7910-11e6-93a5-72374837a78d.png)
![loss_cnn32p-32p-32p-32p_adam1e-4_5000](https://cloud.githubusercontent.com/assets/8138843/18455542/3b14ebee-7910-11e6-873b-ca2a4dc3dfe7.png)

It also seems that the validation loss will keep going up if I train the model for more epochs. Does anyone have idea what's going on here? 

Thanks a lot!

**Comment by jerheff** [comment]
This indicates that the model is overfitting.  It continues to get better and better at fitting the data that it sees (training data) while getting worse and worse at fitting the data that it does not see (validation data).

**Comment by fish128** [comment]
@jerheff Thanks for your reply. We can say that it's overfitting the training data since the training loss keeps decreasing while validation loss started to increase after some epochs. 
However, both the training and validation accuracy kept improving all the time. How can we explain this?

**Comment by jerheff** [comment]
The graph test accuracy looks to be flat after the first 500 iterations or so.  It can remain flat while the loss gets worse as long as the scores don't cross the threshold where the predicted class changes.

**Comment by fish128** [comment]
@jerheff Thanks so much and that makes sense! 
One more question: What kind of regularization method should I try under this situation?

**Comment by fabienbaradel** [comment]
@fish128 Did you find a way to solve your problem (regularization or other loss function)? Thanks in advance.

**Comment by fatemaaa** [comment]
I am training a deep CNN (using vgg19 architectures on Keras) on my data. I used "categorical_cross entropy" as the loss function. During training, the training loss keeps decreasing and training accuracy keeps increasing slowly until convergence. But the validation loss started increasing while the validation accuracy is not improved.
The curve of loss are shown in the following figure:
![epoch loss_relation](https://cloud.githubusercontent.com/assets/24401730/21601812/2a7f5da2-d1c9-11e6-9a42-e8c7e76db14f.png)

**Comment by TimGranskog** [comment]
Just as jerheff mentioned above it is because the model is overfitting on the training data, thus becoming extremely good at classifying the training data but generalizing poorly and causing the classification of the validation data to become worse. You could solve this by stopping when the validation error starts increasing or maybe inducing noise in the training data to prevent the model from overfitting when training for a longer time.

**Comment by pockeyk** [comment]
l

**Comment by s4sarath** [comment]
Even I am also experiencing the same thing. Validation loss is increasing, and validation accuracy is also increased and after some time ( after 10 epochs ) accuracy starts dropping.

**Comment by jayanti-prasad** [comment]
I this the question is still unanswered. Validation loss increase but validation accuracy also increase.

**Comment by MariamMohamedFawzy** [comment]
"Validation loss increases but validation accuracy also increases." 
Does this indicate that you overfit a class or your data is biased, so you get high accuracy on the majority class while the loss still increases as you are going away from the minority classes?

**Comment by MariamMohamedFawzy** [comment]
> The question is still unanswered. Validation loss increases but validation accuracy also increases.

Does this indicate that you overfit a class or your data is biased, so you get high accuracy on the majority class while the loss still increases as you are going away from the minority classes?

**Comment by rajathpatel23** [comment]
I would like to have a follow-up question on this, what does it mean if the validation loss is fluctuating ? and not monotonically increasing or decreasing ?

**Comment by Shay-repos** [comment]
I think your model was predicting more accurately and less certainly about the predictions

**Comment by TedLLH** [comment]
> I think your model was predicting more accurately and less certainly about the predictions

What does this even mean? I have the same situation where val loss and val accuracy are both increasing.

loss: 0.3972 - acc: 0.8039 - val_loss: 2.7228 - val_acc: 0.7161

And when I tested it with test data (not train, not val), the accuracy is still legit and it even has lower loss than the validation data!

loss: 1.6388 - acc: 0.7253

Very confusing......

**Comment by hxnsam** [comment]
this question is still unanswered i am facing same problem while using ResNet  model on my own data.  
Learning rate:  0.0001
73/73 [==============================] - 9s 129ms/step - loss: 0.1621 - acc: 0.9961 - val_loss: 1.0128 - val_acc: 0.8093

Epoch 00100: val_acc did not improve from 0.80934

how can i improve this i have no idea (validation loss is 1.01128 👎 )

**Comment by sanersbug** [comment]
Who has solved this problem? any one can give some point?

**Comment by s4sarath** [comment]
Check your model loss is implementated correctly. It kind of helped me to
spot a bug.

**Comment by alibabadoufu** [comment]
> I would like to have a follow-up question on this, what does it mean if the validation loss is fluctuating ? and not monotonically increasing or decreasing ?

I experienced the same issue but what I found out is because the validation dataset is much smaller than the training dataset. This causes the validation fluctuate over epochs.

**Comment by llgm** [comment]
我也遇到这个问题

**Comment by scorpioWCT** [comment]
Hello I also encountered a similar problem. My training loss and verification loss are relatively stable, but the gap between the two is about 10 times, and the verification loss fluctuates a little, how to solve

**Comment by melvin138** [comment]
![Figure 2020-08-11 214728](https://user-images.githubusercontent.com/60482083/89922205-59ae1f80-dc1c-11ea-86e5-bb3baabe9af2.png)
![Figure 2020-08-11 214654](https://user-images.githubusercontent.com/60482083/89922213-5adf4c80-dc1c-11ea-99ca-f6ed4e52026e.png)

I have the same problem my training accuracy improves and training loss decreases but my validation accuracy gets flattened and my validation loss decreases to some point and increases at the initial stage of learning say 100 epochs (training for 1000 epochs),
Even though I added L2 regularisation and also introduced a couple of Dropouts in my model I still get the same result.
I am working on a time series data so augmentation is still a challege for me. Can anyone suggest some tips to overcome this? Thanks in advance

**Comment by rryoung98** [comment]
This might be helpful: https://discuss.pytorch.org/t/loss-increasing-instead-of-decreasing/18480/4

**Comment by UvanShankar** [comment]
The model is overfitting the training data.
To solve this problem you can try 
1.Regularization
2.Try to add more add to the dataset or try data augumentation

**Comment by kbpark102** [comment]
I'm experiencing similar problem. I tried regularization and data augumentation.
could you give me advice?

**Comment by JhonSanz** [comment]
How about adding more characteristics to the data (new columns to describe the data)? Could it be a way to improve this? (I'm facing the same scenario)

**Comment by chrisjune** [comment]
I also experienced this. After normalize target(or y) column, validation loss is slowly deceased. 
I recommend you to apply min-max or log normalization.

**Comment by PetroleumEngineer** [comment]
> ![Figure 2020-08-11 214728](https://user-images.githubusercontent.com/60482083/89922205-59ae1f80-dc1c-11ea-86e5-bb3baabe9af2.png) ![Figure 2020-08-11 214654](https://user-images.githubusercontent.com/60482083/89922213-5adf4c80-dc1c-11ea-99ca-f6ed4e52026e.png)
> 
> I have the same problem my training accuracy improves and training loss decreases but my validation accuracy gets flattened and my validation loss decreases to some point and increases at the initial stage of learning say 100 epochs (training for 1000 epochs), Even though I added L2 regularisation and also introduced a couple of Dropouts in my model I still get the same result. I am working on a time series data so data augmentation is still a challege for me. I used 80:20% train:test split. Can anyone suggest some tips to overcome this? Thanks in advance

Hello, I am facing exactly the same issue, were you able to solve it >?
![image](https://github.com/keras-team/keras/assets/86235779/992b1c08-674d-4f84-b2d0-dd6d832cf6e2)

**Comment by sondosaabed** [comment]
Hello,
Did anyone fix this issue?
